src/sickle.ts

```diff
@@ -58,10 +62,49 @@ class Annotator {
         this.maybeVisitType((<ts.VariableDeclaration>node).type);
         this.writeNode(node);
         break;
+      case ts.SyntaxKind.ClassDeclaration: {
+        let classNode = <ts.ClassDeclaration>node;
+        let hasCtor = classNode.members.some((e) => e.kind === ts.SyntaxKind.Constructor);
+        if (hasCtor) {
+          this.writeNode(classNode);
+          break;
+        }
+        // Emit a synthetic ctor.
+        // TODO(martinprobst): Handle inherited parent ctors.
+        this.writeTextBetween(classNode, classNode.getFullStart(), classNode.getLastToken());
```

Comment:
every single call in this cl passes `writeTextBetween(node, node.getFullStart(), child)` maybe create a helped `writeTextUntilChild(node, child)`